bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/discovery/HDPowerViewHubMDNSDiscoveryParticipant.java

```diff
@@ -71,14 +67,20 @@ public String getServiceType() {
     public @Nullable DiscoveryResult createResult(ServiceInfo service) {
         for (String host : service.getHostAddresses()) {
             if (VALID_IP_V4_ADDRESS.matcher(host).matches()) {
-                ThingUID thingUID = new ThingUID(THING_TYPE_HUB, host.replace('.', '_'));
-                String generation = this.getGeneration(host);
-                DiscoveryResult hub = DiscoveryResultBuilder.create(thingUID)
-                        .withProperty(HDPowerViewHubConfiguration.HOST, host)
-                        .withRepresentationProperty(HDPowerViewHubConfiguration.HOST)
-                        .withLabel(String.format("@text/%s [\"%s\", \"%s\"]", LABEL_KEY_HUB, generation, host)).build();
-                logger.debug("mDNS discovered Gen {} hub on host '{}'", generation, host);
-                return hub;
+                try {
+                    String serial = propertyGetter.getSerialNumberApiV1(host);
+                    String generation = propertyGetter.getGenerationApiV1(host);
```

Comment:
Regarding:
> However since the prior code did not throw an exception but rather returned a default value, we will need to check once more if the Gen 1 hubs really do not throw an exception here.

To resolve this/remove the doubt, we can simply revert to the old fallback behavior by wrapping it into its own try/catch:
```suggestion
                String generation;
                try {
                    generation = propertyGetter.getGenerationApiV1(host);
                } catch (HubException e) {
                    generation = "1/2";
                }
                try {
                    String serial = propertyGetter.getSerialNumberApiV1(host);
```